### PR TITLE
Lint files

### PR DESCRIPTION
**Pull request by aljones15**
Lints the tests to pass the linter and removes unused variables and comments out unused functions.

I disabled the max-length rule in this project as concatenating all of those test files down is always an issue.

### CODE REVIEW

**Comment by JSAssassin** [comment]
@aljones15 Can you please add the GH actions for linting?

**Comment by aljones15** [comment]
> @aljones15 Can you please add the GH actions for linting?

good idea adding it

File: .eslintrc.cjs [hunk 1]
```diff
@@ -12,6 +12,7 @@ module.exports = {
     sourceType: 'module'
   },
   rules: {
-    'jsdoc/check-examples': 0
+    'jsdoc/check-examples': 0,
+    'max-len': 0
```
Comment (JSAssassin) [inline .eslintrc.cjs:16]: I am a -1 to disabling the max-length rule for the entire project. Instead, we should selectively disable it for specific lines when necessary and try to fix the issues wherever possible.

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+    # update the integration suite once per week at Sunday 5am UTC
+    - cron: '0 5 * * 0'
+
+jobs:
+  lint:
+    if: ${{ github.event_name == 'push' }}
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        node-version: [16.x]
```
Comment (davidlehn) [inline .github/workflows/main.yml:16]: Run all the tooling on `20.x` if possible.

File: .eslintrc.cjs (hunk 1 above)
Comment (aljones15) [reply .eslintrc.cjs]: linted all 98 of them (90 minutes of work to do that) here: https://github.com/digitalbazaar/vc-data-model-2-test-suite/pull/6/commits/5db71923170f8358aa5dcb1cf05c194146dd2bdc

File: .github/workflows/main.yml (hunk 2 above)
Comment (aljones15) [reply .github/workflows/main.yml]: Addressed here: https://github.com/digitalbazaar/vc-data-model-2-test-suite/pull/6/commits/2ad83142ae145d3dbc1b214cfa869cb074988d59

File: .github/workflows/main.yml [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+    steps:
+      - uses: actions/checkout@v2
```
[the remark this answers is not in the document]
Comment (aljones15) [reply .github/workflows/main.yml]: Addressed here: https://github.com/digitalbazaar/vc-data-model-2-test-suite/pull/6/commits/2ad83142ae145d3dbc1b214cfa869cb074988d59

File: .eslintrc.cjs (hunk 1 above)
Comment (JSAssassin) [reply .eslintrc.cjs]: Thanks @aljones15! Sorry for the trouble 😁

File: tests/10-vcdm2.js [hunk 4]
```diff
@@ -147,29 +141,38 @@ describe('Verifiable Credentials Data Model v2.0', function() {
...
-      it.skip('Conforming document (compliance): VCDM "MUST be enforced." ("all relevant normative statements in Sections 4. Basic Concepts, 5. Advanced Concepts, and 6. Syntaxes")', async function() {
+      it.skip('Conforming document (compliance): VCDM "MUST be enforced." ' +
+        '("all relevant normative statements in Sections 4. Basic Concepts, ' +
+        '5. Advanced Concepts, and 6. Syntaxes")', async function() {
       });
       // Basic
-      it.skip('"A serialization format for the conforming document MUST be deterministic, bi-directional, and lossless as described in Section 6. Syntaxes."', async function() {
+      it.skip('"A serialization format for the conforming document MUST be ' +
+        'deterministic, bi-directional, and lossless as described in Section' +
+        ' 6. Syntaxes."', async function() {
```
Comment (JSAssassin) [inline tests/10-vcdm2.js:151]:
```suggestion
      it.skip('A serialization format for the conforming document MUST be ' +
        'deterministic, bi-directional, and lossless as described in Section ' +
        '6. Syntaxes.', async function() {
```

File: tests/10-vcdm2.js [hunk 5]
```diff
@@ -147,29 +141,38 @@ describe('Verifiable Credentials Data Model v2.0', function() {
...
       });
-      it2('"Conforming processors MUST produce errors when nonconforming documents are consumed."', async function() {
+      it2('"Conforming processors MUST produce errors when nonconforming ' +
+        'documents are consumed."', async function() {
```
Comment (JSAssassin) [inline tests/10-vcdm2.js:154]:
```suggestion
      it2('Conforming processors MUST produce errors when nonconforming ' +
        'documents are consumed.', async function() {
```

File: tests/10-vcdm2.js [hunk 6]
```diff
@@ -147,29 +141,38 @@ describe('Verifiable Credentials Data Model v2.0', function() {
...
+          type: ['NonconformingDocument']
         };
         await assert.rejects(issue(doc));
         await assert.rejects(verify(doc));
         await assert.rejects(prove(doc));
         await assert.rejects(verifyVp(doc));
       });
-      it2('"Verifiable credentials and verifiable presentations MUST include a @context property."', async function() {
+      it2('"Verifiable credentials and verifiable presentations MUST include' +
+        ' a @context property."', async function() {
```
Comment (JSAssassin) [inline tests/10-vcdm2.js:164]:
```suggestion
      it2('Verifiable credentials and verifiable presentations MUST include ' +
        'a @context property.', async function() {
```

File: tests/10-vcdm2.js [hunk 7]
```diff
@@ -147,29 +141,38 @@ describe('Verifiable Credentials Data Model v2.0', function() {
...
         await assert.rejects(issue(
           require('./input/credential-no-context-fail.json')));
         await assert.rejects(prove(
           require('./input/presentation-no-context-fail.json')));
       });
-      it2('Verifiable credentials and verifiable presentations: "The value of the @context property MUST be an ordered set where the first item is a URL with the value https://www.w3.org/ns/credentials/v2."', async function() {
+      it2('Verifiable credentials and verifiable presentations: ' +
+        '"The value of the @context property MUST be an ordered set where ' +
+        'the first item is a URL with the value https://www.w3.org/ns/' +
+        'credentials/v2."', async function() {
```
Comment (JSAssassin) [inline tests/10-vcdm2.js:175]:
```suggestion
        'The value of the @context property MUST be an ordered set where ' +
        'the first item is a URL with the value https://www.w3.org/ns/' +
        'credentials/v2.', async function() {
```

File: tests/10-vcdm2.js [hunk 8]
```diff
@@ -190,39 +195,49 @@ describe('Verifiable Credentials Data Model v2.0', function() {
         await assert.rejects(
           issue(require('./input/credential-context-combo5-fail.json')));
       });
-      it2('"All libraries or processors MUST ensure that the order of the values in the @context property is what is expected for the specific application."', async function() {
+      it2('"All libraries or processors MUST ensure that the order of the ' +
+        'values in the @context property is what is expected for the ' +
+        'specific application."', async function() {
```
Comment (JSAssassin) [inline tests/10-vcdm2.js:200]:
```suggestion
      it2('All libraries or processors MUST ensure that the order of the ' +
        'values in the @context property is what is expected for the ' +
        'specific application.', async function() {
```

File: tests/10-vcdm2.js [hunk 9]
```diff
@@ -190,39 +195,49 @@ describe('Verifiable Credentials Data Model v2.0', function() {
...
+        'MAY be dereferenced."', async function() {
         await assert.rejects(
           issue(require('./input/credential-not-url-id-fail.json')));
       });
-      it2('"The value of the id property MUST be a single URL."', async function() {
-        await assert.rejects(
-          issue(require('./input/credential-nonsingle-id-fail.json')));
-      });
-      it2('"Verifiable credentials and verifiable presentations MUST have a type property."', async function() {
+      it2('"The value of the id property MUST be a single URL."',
```
Comment (JSAssassin) [inline tests/10-vcdm2.js:226]:
```suggestion
      it2('The value of the id property MUST be a single URL.',
```

File: tests/10-vcdm2.js [hunk 10]
```diff
@@ -190,39 +195,49 @@ describe('Verifiable Credentials Data Model v2.0', function() {
...
+        async function() {
+          await assert.rejects(
+            issue(require('./input/credential-nonsingle-id-fail.json')));
+        });
+      it2('"Verifiable credentials and verifiable presentations MUST have ' +
+        'a type property."', async function() {
```
Comment (JSAssassin) [inline tests/10-vcdm2.js:232]:
```suggestion
      it2('Verifiable credentials and verifiable presentations MUST have ' +
        'a type property.', async function() {
```

File: tests/10-vcdm2.js [hunk 11]
```diff
@@ -190,39 +195,49 @@ describe('Verifiable Credentials Data Model v2.0', function() {
...
         await assert.rejects(
           issue(require('./input/credential-no-type-fail.json')));
         await assert.rejects(
           prove(require('./input/presentation-no-type-fail.json')));
       });
-      it2('"The value of the type property MUST be, or map to (through interpretation of the @context property), one or more URLs."', async function() {
+      it2('"The value of the type property MUST be, or map to (through ' +
+        'interpretation of the @context property), one or more URLs."',
+      async function() {
```
Comment (JSAssassin) [inline tests/10-vcdm2.js:240]:
```suggestion
      it2('The value of the type property MUST be, or map to (through ' +
        'interpretation of the @context property), one or more URLs.',
      async function() {
```

File: tests/10-vcdm2.js [hunk 12]
```diff
@@ -255,37 +272,45 @@ describe('Verifiable Credentials Data Model v2.0', function() {
...
-      });
-      it2('"A verifiable credential MUST have a credentialSubject property."', async function() {
-        await assert.rejects(
-          issue(require('./input/credential-no-subject-fail.json')));
-      });
-      it2('"The value of the credentialSubject property is defined as a set of objects that MUST contain one or more claims that are each related to a subject of the verifiable credential."', async function() {
+      it.skip('"All credentials, presentations, and encapsulated objects ' +
+        'MUST specify, or be associated with, additional more narrow types ' +
+        '(like UniversityDegreeCredential, for example) so software systems' +
+        ' can process this additional information."', async function() {
```
Comment (JSAssassin) [inline tests/10-vcdm2.js:289]:
```suggestion
      it.skip('All credentials, presentations, and encapsulated objects ' +
        'MUST specify, or be associated with, additional more narrow types ' +
        '(like UniversityDegreeCredential, for example) so software systems ' +
        ' can process this additional information.', async function() {
```

File: tests/10-vcdm2.js [hunk 13]
```diff
@@ -255,37 +272,45 @@ describe('Verifiable Credentials Data Model v2.0', function() {
...
+      });
+      it2('"A verifiable credential MUST have a credentialSubject property."',
+        async function() {
```
Comment (JSAssassin) [inline tests/10-vcdm2.js:292]:
```suggestion
      it2('A verifiable credential MUST have a credentialSubject property.',
        async function() {
```

File: tests/10-vcdm2.js [hunk 14]
```diff
@@ -255,37 +272,45 @@ describe('Verifiable Credentials Data Model v2.0', function() {
...
+          await assert.rejects(
+            issue(require('./input/credential-no-subject-fail.json')));
+        });
+      it2('"The value of the credentialSubject property is defined as a set' +
+        ' of objects that MUST contain one or more claims that are each ' +
+        'related to a subject of the verifiable credential."',
+      async function() {
```
Comment (JSAssassin) [inline tests/10-vcdm2.js:299]:
```suggestion
      it2('The value of the credentialSubject property is defined as a set ' +
        'of objects that MUST contain one or more claims that are each ' +
        'related to a subject of the verifiable credential.',
      async function() {
```

File: tests/10-vcdm2.js [hunk 15]
```diff
@@ -255,37 +272,45 @@ describe('Verifiable Credentials Data Model v2.0', function() {
...
-      it2('"A verifiable credential MUST have an issuer property."', async function() {
-        await assert.rejects(
-          issue(require('./input/credential-no-issuer-fail.json')));
-      });
-      it2('"The value of the issuer property MUST be either a URL or an object containing an id property."', async function() {
+          issue(require(
+            './input/credential-subject-multiple-empty-fail.json')));
+      });
+      it2('"A verifiable credential MUST have an issuer property."',
+        async function() {
```
Comment (JSAssassin) [inline tests/10-vcdm2.js:308]:
```suggestion
      it2('A verifiable credential MUST have an issuer property.',
        async function() {
```

File: tests/10-vcdm2.js [hunk 16]
```diff
@@ -255,37 +272,45 @@ describe('Verifiable Credentials Data Model v2.0', function() {
...
+          await assert.rejects(
+            issue(require('./input/credential-no-issuer-fail.json')));
+        });
+      it2('"The value of the issuer property MUST be either a URL or an' +
+        ' object containing an id property."', async function() {
```
Comment (JSAssassin) [inline tests/10-vcdm2.js:313]:
```suggestion
      it2('The value of the issuer property MUST be either a URL or an ' +
        'object containing an id property.', async function() {
```

**Review by JSAssassin — APPROVED** [review]
Thanks approving, there are just some minor nits but looks good.

File: .eslintrc.cjs (hunk 1 above)
Comment (aljones15) [reply .eslintrc.cjs]: not any major trouble, but we do have a deadline so we might want to keep tasks a little more focused on the main goal of getting these test suites working.

File: tests/10-vcdm2.js (hunk 4 above)
Comment (aljones15) [reply tests/10-vcdm2.js:151]: this is basically everywhere we might want to just do all of this one commit and not via github comments.

File: example-server.js [hunk 17]
```diff
@@ -1,11 +1,16 @@
-import {createServer} from 'http'
+/*!
+ * Copyright (c) 2022 Digital Bazaar, Inc. All rights reserved.
```
[the remark this answers is not in the document]
Comment (aljones15) [reply example-server.js]: I bumped the copyrights.

**Pull request merged**